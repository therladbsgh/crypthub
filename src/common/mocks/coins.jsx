export const coins = [
    {
        name: 'Bitcoin',
        symbol: 'BTC',
        currPrice: 10000
    },
    {
        name: 'Ethereum',
        symbol: 'ETH',
        currPrice: 5000
    }
]
