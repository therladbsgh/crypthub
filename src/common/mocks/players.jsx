import date from 'date-and-time';

export const player1 = {
    name: 'player1',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 250
        },
        {
            name: 'Ethereum',
            amount: 1000.28373623
        },
        {
            name: 'USD',
            amount: 250
        }
    ],
    transactions: {
        history: [
            {
                id: '1',
                type: 'market',
                side: 'buy',
                size: 25,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 1 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: true,
                filledDate: date.parse('Apr 3 2018', 'MMM D YYYY')
            },
            {
                id: '2',
                type: 'limit',
                side: 'buy',
                size: 35.000006,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('Apr 5 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ],
        current: [
            {
                id: '3',
                type: 'short',
                side: 'sell',
                size: 80,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: false,
                filledDate: null
            },
            {
                id: '4',
                type: 'limit',
                side: 'buy',
                size: 2,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 7 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('May 2 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ]
    },
    netWorth: 200,
    numTrades: 80,
    netReturn: -20,
    todayReturn: 5,
    currRank: 2,
    buyingPower: 150,
    shortReserve: 50
};

export const player2 = {
    name: 'player2',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 300
        }
    ],
    transactions: {
        history: [
            {
                id: '5',
                side: 'buy',
                size: 1,
                price: 10,
                symbol: 'ETH'
            },
            {
                id: '6',
                side: 'sell',
                size: 3500,
                price: 202220,
                symbol: 'ETH'
            }
        ],
        current: [
            {
                id: '7',
                side: 'sell',
                size: 23230,
                price: 232,
                symbol: 'BTC'
            },
            {
                id: '8',
                side: 'buy',
                size: 6000,
                price: 50,
                symbol: 'ETH'
            }
        ]
    },
    netWorth: 300,
    numTrades: 2,
    netReturn: 120
};
