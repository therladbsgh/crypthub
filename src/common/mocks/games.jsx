import date from 'date-and-time';
import * as PlayerMocks from './players';

export const game1 = {
    id: '1',
    name: 'game1',
    description: 'desc1',
    host: 'user1',
    created: date.parse('Apr 1 2018', 'MMM D YYYY'),
    start: date.parse('Apr 2 2018', 'MMM D YYYY'),
    end: date.parse('May 7 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game2 = {
    id: '2',
    name: 'game2',
    description: 'desc2',
    host: 'user2',
    created: date.parse('Apr 2 2018', 'MMM D YYYY'),
    start: date.parse('Apr 3 2018', 'MMM D YYYY'),
    end: date.parse('May 1 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game3 = {
    id: '3',
    name: 'game3',
    description: 'desc3',
    host: 'user3',
    created: date.parse('Apr 4 2018', 'MMM D YYYY'),
    start: date.parse('Apr 8 2018', 'MMM D YYYY'),
    end: date.parse('Apr 28 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};
