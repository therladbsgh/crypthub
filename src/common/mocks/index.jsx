import * as GameMocks from './games';
import * as PlayerMocks from './players';

export { GameMocks, PlayerMocks };
