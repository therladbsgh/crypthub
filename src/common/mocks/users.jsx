import * as GameMocks from './games';

export const user1 = {
    id: 'id1',
    name: 'user1',
    ELO: 3000,
    games: [
        GameMocks.game1,
        GameMocks.game2
    ]
};

export const user2 = {
    id: 'id2',
    name: 'user2',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};

export const user3 = {
    id: 'id3',
    name: 'user3',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};
