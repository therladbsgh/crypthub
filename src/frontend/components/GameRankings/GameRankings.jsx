import * as _ from 'lodash';
import React, { Component } from 'react';
import formatCurrency from 'format-currency';
import { Header, Table, Icon, Pagination } from 'semantic-ui-react';
import { Searchbar, GameCard } from 'components';
import { SharedStyle as styles } from 'styles';

const numPerPage = 10;

export default class GameRankings extends Component {
    constructor(props) {
		super(props);
		this.state = {
			activePage: 1
		};

		this.handlePageChange = this.handlePageChange.bind(this);
    }

    handlePageChange(event, { activePage }) {
		this.setState({ activePage });
    }

    render() {
        const { activePage } = this.state;

        const players = _.sortBy(this.props.players, ({ netWorth }) => -netWorth); // sort by currRank?

        const upper = activePage * numPerPage;
        const playersShown = _.slice(players, (activePage - 1) * numPerPage, upper > players.length ? players.length : upper);
        const totalPages = Math.ceil(players.length / numPerPage);

        return (
            <div>
                <Header as='h2'>Rankings</Header>
                <Table celled>
                    <Table.Header>
                        <Table.Row>
                            <Table.HeaderCell><Icon name='trophy' /></Table.HeaderCell>
                            <Table.HeaderCell>Name</Table.HeaderCell>
                            <Table.HeaderCell>Net Worth</Table.HeaderCell>
                            <Table.HeaderCell>Trades</Table.HeaderCell>
                            <Table.HeaderCell>Net Return</Table.HeaderCell>
                        </Table.Row>
                    </Table.Header>

                    <Table.Body>
                        {_.map(playersShown, (p, index) => 
                            <Table.Row key={index}>
                                <Table.Cell>{p.currRank}</Table.Cell>
                                <Table.Cell>{p.username}</Table.Cell>
                                <Table.Cell>{formatCurrency(p.netWorth, { format: '%s%v', symbol: '$' })}</Table.Cell>
                                <Table.Cell>{p.numTrades}</Table.Cell>
                                <Table.Cell className={p.netReturn >= 0 ? styles.pos : styles.neg}>{formatCurrency(Math.abs(p.netReturn), { format: '%s%v', symbol: `${p.netReturn >= 0 ? '+' : '-'}$` })}</Table.Cell>
                            </Table.Row>
                        )}
                    </Table.Body>
                </Table>
                {totalPages > 1 &&
                <div key='2' className={styles.center}>
                    <Pagination totalPages={totalPages} activePage={activePage} onPageChange={this.handlePageChange} />
                </div>}
            </div>
        );
    }
}
