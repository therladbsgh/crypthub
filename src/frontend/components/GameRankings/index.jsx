import GameRankings from './GameRankings';

export default GameRankings;
