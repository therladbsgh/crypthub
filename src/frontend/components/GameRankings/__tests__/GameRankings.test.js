import React from 'react';
import GameRankings from '../GameRankings';
import { GameMocks } from 'mocks';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  	const tree = renderer.create(<GameRankings player={GameMocks.game1.players} />).toJSON();
  	expect(tree).toMatchSnapshot();
});