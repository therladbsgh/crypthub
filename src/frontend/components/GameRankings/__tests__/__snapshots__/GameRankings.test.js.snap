// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2
    className="ui header"
  >
    Rankings
  </h2>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          <i
            aria-hidden="true"
            className="trophy icon"
          />
        </th>
        <th
          className=""
        >
          Name
        </th>
        <th
          className=""
        >
          Net Worth
        </th>
        <th
          className=""
        >
          Trades
        </th>
        <th
          className=""
        >
          Net Return
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    />
  </table>
</div>
`;
